packages/client/src/data_commons_client.ts

```diff
@@ -372,9 +391,52 @@ class DataCommonsClient {
       populationObservations
     );
 
+    // Filter by start/end date if specified
+    if (params.startDate || params.endDate) {
+      return Promise.resolve(
+        dataRows.filter((dataRow) => {
+          const observationDate = dataRow.variable.observation.date;
+          if (!observationDate) {
+            return false;
+          }
+          if (params.startDate && params.startDate > observationDate) {
+            return false;
+          }
+          if (params.endDate && params.endDate < observationDate) {
+            return false;
+          }
+          return true;
+        })
+      );
+    }
+
     return Promise.resolve(dataRows);
   }
 
+  /**
+   * Overrides any of the passed in facets with values from statMetadataOverride
+   * @param facets
+   * @returns facets with overriden values
+   */
+  overrideFacetValues(facets: FacetStore): FacetStore {
+    if (!this.facetOverride || Object.keys(this.facetOverride).length === 0) {
+      return facets;
+    }
+    const newFacets = {
+      ...facets,
+    };
+    Object.keys(newFacets).forEach((facetId) => {
+      const facet = newFacets[facetId];
+      if (facet.unit && facet.unit in this.facetOverride) {
```

Comment:
also, are we only overriding units? if that's the case, then maybe be more specific in the naming/description of this function

Reply:
It's overriding the entire facet / StatMetadata, but we know which values to override based on the facet's unit (since the FacetStore is keyed by facetId, which isn't a stable key for overriding)

Reply:
Updated the comment to hopefully make this more clear